### PR TITLE
[Linux][A11y] implement AtkEditableText interface

### PR DESCRIPTION
**Pull request by jpnurmi**
This PR implements the [AtkEditableText](https://gnome.pages.gitlab.gnome.org/atk/AtkEditableText.html) interface to add full accessibility support for Flutter text fields on Linux.

Solves: flutter/flutter#103191

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

https://user-images.githubusercontent.com/140617/173051820-1de52ad8-5771-46ae-b20f-3d6b7b8f68fd.mp4

### CODE REVIEW

File: shell/platform/linux/fl_view_accessible.cc [hunk 1]
```diff
@@ -36,40 +41,93 @@ static FlEngine* get_engine(FlViewAccessible* self) {
   return self->engine;
 }
 
+static FlAccessibleNode* create_node(FlViewAccessible* self,
+                                     const FlutterSemanticsNode* semantics) {
+  FlAccessibleNode* node = nullptr;
+  FlEngine* engine = get_engine(self);
+  if (semantics->flags & kFlutterSemanticsFlagIsTextField) {
+    node = fl_accessible_text_field_new(engine, semantics->id);
```
Comment (robert-ancell) [inline shell/platform/linux/fl_view_accessible.cc:49]: Might be simpler/safer to just return the value here rather than potentially leak it later if this function was changed in the future to return after the if branch.

File: shell/platform/linux/fl_accessible_text_field.cc [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+                                             FlutterSemanticsAction action,
+                                             GBytes* data) {
+  switch (action) {
+    case kFlutterSemanticsActionMoveCursorForwardByCharacter:
+    case kFlutterSemanticsActionMoveCursorBackwardByCharacter:
+    case kFlutterSemanticsActionMoveCursorForwardByWord:
+    case kFlutterSemanticsActionMoveCursorBackwardByWord: {
+      g_autoptr(FlValue) extend_selection = fl_value_new_bool(false);
+      g_autoptr(FlStandardMessageCodec) codec = fl_standard_message_codec_new();
+      data = fl_message_codec_encode_message(FL_MESSAGE_CODEC(codec),
```
Comment (robert-ancell) [inline shell/platform/linux/fl_accessible_text_field.cc:123]: This is going to leak - you need a g_autoptr(GBytes) for this value that overrides the data passed in.

File: shell/platform/linux/fl_accessible_text_field.cc [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+                                                  n_chars, chars, nullptr);
+                          }),
+                          self, G_CONNECT_SWAPPED);
+}
+
+FlAccessibleNode* fl_accessible_text_field_new(FlEngine* engine, int32_t id) {
+  FlAccessibleNode* self =
+      FL_ACCESSIBLE_NODE(g_object_new(fl_accessible_text_field_get_type(),
+                                      "engine", engine, "id", id, nullptr));
+  return self;
```
Comment (robert-ancell) [inline shell/platform/linux/fl_accessible_text_field.cc:397]: Nit: could just directly return rather than assigning to self.

File: shell/platform/linux/fl_accessible_node.cc [hunk 4]
```diff
@@ -46,7 +47,6 @@ static ActionData action_mapping[] = {
      "MoveCursorForwardByCharacter"},
     {kFlutterSemanticsActionMoveCursorBackwardByCharacter,
      "MoveCursorBackwardByCharacter"},
-    {kFlutterSemanticsActionSetSelection, "SetSelection"},
```
Comment (robert-ancell) [inline shell/platform/linux/fl_accessible_node.cc:49]: Why is this removed?

**Review by robert-ancell — CHANGES_REQUESTED** [review]
Other than the above LGTM

File: shell/platform/linux/fl_accessible_node.cc (hunk 4 above)
Comment (jpnurmi) [reply shell/platform/linux/fl_accessible_node.cc:49]: The SetSelection action cannot be executed "standalone" because it requires the target selection as an [argument](https://github.com/flutter/flutter/blob/712860d33b00bf3acf02118512901a6261ad5219/packages/flutter/lib/src/semantics/semantics.dart#L3608-L3616). Performing the action would result to:
```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: 'package:flutter/src/semantics/semantics.dart': Failed assertion: line 3609 pos 14: 'args != null && args is Map': is not true.
#0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
#1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
#2      SemanticsConfiguration.onSetSelection=.<anonymous closure> (package:flutter/src/semantics/semantics.dart:3609:14)
#3      SemanticsOwner.performAction (package:flutter/src/semantics/semantics.dart:3131:14)
#4      RendererBinding._handleSemanticsAction (package:flutter/src/rendering/binding.dart:362:36)
#5      _invoke3 (dart:ui/hooks.dart:207:13)
#6      PlatformDispatcher._dispatchSemanticsAction (dart:ui/platform_dispatcher.dart:1030:5)
#7      _dispatchSemanticsAction (dart:ui/hooks.dart:99:31)
```

Removing the action seemed reasonable given that `FlAccessibleTextField` implements `AtkText::set_selection` and [AtkAction](https://gnome.pages.gitlab.gnome.org/atk/AtkAction.html) docs say:
> Exceptions: when the user interaction is already covered by another appropriate interface such as [AtkEditableText](https://gnome.pages.gitlab.gnome.org/atk/AtkEditableText.html) (insert/delete text, etc.) or [AtkValue](https://gnome.pages.gitlab.gnome.org/atk/AtkValue.html) (set value) then these actions should not be exposed by [AtkAction](https://gnome.pages.gitlab.gnome.org/atk/AtkAction.html) as well.

**Review by robert-ancell — APPROVED** [review]
LGTM

**Comment by chinmaygarde** [comment]
From Triage: Is this blocked? May I commit this?
